### PR TITLE
Playlist global/player stat filtering

### PR DESCRIPTION
**Pull request by Sciguymjm**
We'll default to 13 in the meantime, until the frontend is updated.

I envision this to be a simple dropdown on the player page that has the current playlists in it, and switching them updates the spider charts. The other thing I would like is to default to whatever they play the most (or make this an option on login).

### CODE REVIEW

File: backend/blueprints/spa_api/service_layers/player/player_ranks.py [hunk 1]
```diff
@@ -24,10 +24,26 @@ def __init__(self, ranks: Dict[str, PlaylistRank]):
             self.solo = ranks['solo'].__dict__
         else:
             self.solo = PlaylistRank("Unranked (div 1)", 0, 0).__dict__
-        if 'standard' in ranks and  ranks['standard'] is not None:
+        if 'standard' in ranks and ranks['standard'] is not None:
             self.standard = ranks['standard'].__dict__
         else:
             self.standard = PlaylistRank("Unranked (div 1)", 0, 0).__dict__
+        if 'hoops' in ranks and ranks['hoops'] is not None:
```
Comment (dtracers) [inline backend/blueprints/spa_api/service_layers/player/player_ranks.py]: we should add a github issue to change this stuff into a list

File: backend/database/wrapper/rank_wrapper.py [hunk 2]
```diff
@@ -1,24 +1,44 @@
-from backend.utils.psyonix_api_handler import get_rank_batch
+from backend.database.objects import Playlist
 
+rank_mapping = {
```
Comment (dtracers) [inline backend/database/wrapper/rank_wrapper.py]: can you add a comment for the purpose of this?

File: backend/database/wrapper/stats/global_stats_wrapper.py [hunk 3]
```diff
@@ -41,29 +60,35 @@ def float_maybe(f):
...
-            results[column.get_field_name()] = column_results
-        return results
+        playlist_results = {}
+        for playlist in GLOBAL_STATS_PLAYLISTS:
+            results = {}
+            for column, q in zip(self.stat_list, self.stats_query):
+                column_results = []
+                # set the column result
+                self.base_query.clean().with_stat_query([PlayerGame.player, q.label('avg')])
+                for rank in ranks:
```
Comment (dtracers) [inline backend/database/wrapper/stats/global_stats_wrapper.py]: a triple for loop :O
can we put this most inner stuff into its own method?

File: backend/database/wrapper/stats/player_stat_wrapper.py [hunk 4]
```diff
@@ -42,30 +42,50 @@ def get_wrapped_stats(self, stats):
...
+        """
+        Gets the averaged stats for the given ID, in either raw or standard deviation form.
+
+        :param session: DBSession
+        :param id_: Player ID to retrieve
+        :param rank: rank in integer form (0-19) (optional)
+        :param redis: optional redis instance, for caching/retrieving
+        :param raw: return the values in raw form instead of compared to global
+        :param replay_ids: replay ids to get stats from
+        :param playlist: what playlist to filter by (only when replay_ids is None)
```
Comment (dtracers) [inline backend/database/wrapper/stats/player_stat_wrapper.py]: can you add win?

File: backend/tasks/celery_tasks.py [hunk 5]
```diff
@@ -94,7 +94,7 @@ def better_json_dumps(response: object):
 
 @celery.on_after_configure.connect
 def setup_periodic_tasks(sender, **kwargs):
-    sender.add_periodic_task(10 * 60, calc_global_stats.s(), name='calculate global stats every 10 min')
+    sender.add_periodic_task(60 * 60 * 3, calc_global_stats.s(), name='calculate global stats every 10 min')
```
Comment (dtracers) [inline backend/tasks/celery_tasks.py]: can we make it so that the name always matches the time?

File: backend/utils/psyonix_api_handler.py [hunk 6]
```diff
@@ -97,29 +99,30 @@ def get_rank_batch(ids, offline_redis=None):
     if 'detail' in data:
         return {str(i): {} for i in ids}
     for player in data:
-        rank_datas = []
+        rank_datas = {}
         unique_id = player['user_id']
-        names = {'13': 'standard', '11': 'doubles', '10': 'duel', '12': 'solo'}
+        names = {'13': 'standard', '11': 'doubles', '10': 'duel', '12': 'solo',
```
Comment (dtracers) [inline backend/utils/psyonix_api_handler.py]: We have enums that should already contain this mapping

File: backend/utils/psyonix_api_handler.py [hunk 7]
```diff
@@ -199,5 +202,6 @@ def get_platform_id(i):
     else:
         return '-1'
 
+
 if __name__ == '__main__':
-    print(get_rank('76561198374703623'))
\ No newline at end of file
+    print(get_rank_batch(['76561198374703623'], use_redis=False))
```
Comment (dtracers) [inline backend/utils/psyonix_api_handler.py]: move to a test file?

File: webapp/src/Components/Player/Overview/PlayStyle/PlayStyleActions.tsx [hunk 8]
```diff
@@ -4,23 +4,26 @@ import * as React from "react"
...
 interface State {
     dialogOpen: boolean
+    playlist: number
 }
 
 export class PlayStyleActions extends React.PureComponent<Props, State> {
     constructor(props: Props) {
         super(props)
-        this.state = {dialogOpen: false}
+        this.state = {dialogOpen: false, playlist: 13}
```
Comment (dtracers) [inline webapp/src/Components/Player/Overview/PlayStyle/PlayStyleActions.tsx]: can we just change from the enum and use that everywhere?

File: backend/utils/psyonix_api_handler.py (hunk 7 above)
Comment (Sciguymjm) [reply backend/utils/psyonix_api_handler.py]: Nope, that would require putting the RLAPI key in a test file.

File: backend/utils/psyonix_api_handler.py (hunk 6 above)
Comment (Sciguymjm) [reply backend/utils/psyonix_api_handler.py]: We don't have human-readable strings.

Comment (Sciguymjm) [reply backend/utils/psyonix_api_handler.py]: Okay, but thats not required for this PR. We can do a PR where we change this globally.

File: webapp/src/Components/Player/PlayerOverview.tsx [hunk 9]
```diff
@@ -22,12 +22,13 @@ const playerViewTabs = ["Profile", "Playstyle", "Match History"]
...
 interface State {
     selectedMobileTab?: PlayerViewTab
+    playlist?: number
 }
 
 class PlayerOverviewComponent extends React.PureComponent<Props, State> {
     constructor(props: Props) {
         super(props)
-        this.state = {selectedMobileTab: "Profile"}
+        this.state = {selectedMobileTab: "Profile", playlist: 3}
```
[the remark this answers is not in the document]
Comment (Sciguymjm) [reply webapp/src/Components/Player/PlayerOverview.tsx]: Default playlist (standard, I will change it)

**Pull request merged**